File: TravellerCreator/ClientApp/src/app/sector-creator/models/mongoose/mongoose-sector.ts
import {ISector} from "../interfaces/sector";
import {MongooseSubsector} from "./mongoose-subsector";
import {SectorType} from "../enums/sector-type";

export class MongooseSector implements ISector {
  sectorType: SectorType = SectorType.Basic;
  subsectors: MongooseSubsector[] = [];
}

File: TravellerCreatorModels/Mongoose/MongooseSector.cs
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.Mongoose;

public class MongooseSector : ISector
{
    public SectorType SectorType => SectorType.Basic;
    public List<ISubsector> Subsectors { get; set; } = new();
}